### PR TITLE
Adding GitHub authentication

### PR DESCRIPTION
**Pull request by dckesler**
### Description

Adds github authentication flow

### CODE REVIEW

File: apps/web/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import NextAuth, { AuthOptions, Profile, Session } from 'next-auth'
```
Comment (nicolasbrugneaux) [inline apps/web/pages/api/auth/[...nextauth].ts:1]: filename is strange, is this correct?

File: apps/web/pages/index.tsx [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 import styles from 'styles/Home.module.css'
+
 export const inter = Inter({ subsets: ['latin'] })
 
 interface Props {
   isOutOfCELO: boolean
 }
 
 export default function Home({isOutOfCELO}: Props) {
+  const { data: session } = useSession()
```
Comment (nicolasbrugneaux) [inline apps/web/pages/index.tsx:18]: is `session` used?

File: apps/web/src/request-form.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ export default function RequestForm({isOutOfCELO}:Props) {
         headers: {
           "Content-Type": "application/json",
         },
-        body: JSON.stringify({beneficiary, captchaToken, skipStables}),
+        body: JSON.stringify({beneficiary, captchaToken, skipStables, network: 'alfajores'}),
```
Comment (nicolasbrugneaux) [inline apps/web/src/request-form.tsx:54]: maybe a constant would be better than a magic string

File: apps/web/pages/api/auth/[...nextauth].ts (hunk 1 above)
Comment (dckesler) [reply apps/web/pages/api/auth/[...nextauth].ts:1]: Yes

Comment (dckesler) [reply apps/web/pages/api/auth/[...nextauth].ts:1]: https://next-auth.js.org/getting-started/example#add-api-route

File: apps/web/public/github_white.png
Comment (aaronmgdr) [inline apps/web/public/github_white.png:1]: a bit bigger than needed

Comment (aaronmgdr) [reply apps/web/public/github_white.png:1]: but i guess byte size isnt crazy

File: apps/web/src/firebase-serverside.ts [hunk 4]
```diff
@@ -37,18 +37,21 @@ async function getDB(): Promise<firebase.database.Database> {
 
 export async function sendRequest(
   beneficiary: Address,
-  skipStables: boolean
+  skipStables: boolean,
+  network: string,
```
Comment (aaronmgdr) [inline apps/web/src/firebase-serverside.ts:41]: i think this should be restricted to just certain strings

File: apps/web/src/request-form.tsx [hunk 5]
```diff
@@ -80,12 +79,13 @@ export default function RequestForm({isOutOfCELO}:Props) {
...
 
   const previousAddress = useLastAddress()
+  const buttonDisabled = !executeRecaptcha || !!faucetRequestKey || disableCELOWhenOut
 
 
   return <>
     <div className={styles.intro}>
         <p className={`${inter.className} ${styles.center}`}>
-          Enter your testnet address below. Each request gives you: {STANDARD_AMOUNT} CELO{!skipStables && `, ${STANDARD_AMOUNT} cUSD, ${STANDARD_AMOUNT} cEUR, & ${STANDARD_AMOUNT} cREAL` }*.
+          Enter your testnet address below. Authenticate with GitHub to receive more tokens.
```
Comment (aaronmgdr) [inline apps/web/src/request-form.tsx:88]: 😍

**Review by aaronmgdr — COMMENTED** [review]
Looks great overall

File: apps/web/src/faucet-interfaces.ts [hunk 6]
```diff
@@ -14,13 +15,19 @@ export enum RequestType {
   Faucet = "Faucet",
 }
 
+export enum AuthLevel {
```
[the remark this answers is not in the document]
Comment (dckesler) [reply apps/web/src/faucet-interfaces.ts:18]: I had the same thought and was really close to adding it but alas

**Pull request merged**